### ISSUE TITLE
Add support for box/rectangle zoom

### ISSUE THREAD

**Issue by tweeter0830** [vega/vega#1715]
I spend a lot of time looking at engineering time-series data.

I currently use bokeh for this, but I would love to switch to vega/vega-lite/altair.

However, I'm hesitant to switch because vega appears to be missing a box-zoom feature. This feature is incredibly useful for exploratory time-series plotting/analysis.

As an example, my data often looks like a more complex version of [this bokeh example.](
http://bokeh.pydata.org/en/latest/docs/gallery/box_annotation.html)

There may be many phenomenon of different x/y axis scales which get manifested in time-series data and being able to change the zoom level/aspect ratio is incredibly useful.

Basically, I would love to see Bokeh's [BoxZoomTool](https://bokeh.pydata.org/en/latest/docs/user_guide/tools.html#boxzoomtool) (as enabled in the box_annotaion example above) in vega/vega-lite/altair.

**Comment by jheer** [comment]
Thanks for posting! The good news is that what you describe is _already_ expressible within Vega. Here is [a modified zoomable scatterplot example](https://vega.github.io/editor/#/url/vega/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6SgFY6AK1jcAdpAA04KAHdGAExbYwogAw7FESCkbV8WgMx6DUQujVrGM6ltHXI6Vnm6xGAL2RakLZqCiBKkD7UMugANrBaANpKEKAQaVDRqAE4kBgATkxy1umQFLGs2WAATDrJYAC+xWCp6RnoWYFq3MpFUGUxFVoyrDExTYayiXXpwJDIVDJ48Tmo3KywyF09ClCshGroeJWQAB4AngAUAJSQ9dNgALp1jXUtJZnHJ-CseTul5ZVhqNxlBJjgkq00m9IVB5shFssoKt1ptukV7iU9gcjoFYDFGPBkBcTl1UDcMQ17k90i8ZvdIB9Amdvr95H0AUMRmN6WCwBCYdDIXMFktAsiNlt0TC0pAsYdjniCUSzqTydK7pDqWlaVD6Yycr5uNxUF4-v1Bjggdyhbz+ZDBa1hfDRTlVDItlhxcg9qFpYY5TicuxuCcwAAfMNgQ3G00UjWtLUQHUpPXtY7Bk5mjmWrkgyC2ikOkpwhFTP268slaQ-QmBN1bX2Vwx4M6EY5e1ZUP4AM0YMSOvxyJbwdFgiEY3bwAGlkGdG03IPBZLBWB0cHg8hUQdLIOxkHhlMh4WWm1DIC222K1hK0T2+wPAsPR+PJzO58nT7Nq3la67HA22XPVt22vb1CFuRM-XjaUHm3GUA2OSUwAAfj5E5rjZS4rgeMAc1GSA4zgosZWHRFIHrbpPVAn04P9fZ5UCK0CPVKlniaB0GTTTpkH7TBAPNSoEn0MAdFgnk5HBQsKUMUiT3LYihW-X8VH-bp50rIDLxWUDOwCQDe37ZBBygABCJ8xwnadZ3U8tF2XVdKg3LdpMdPcDyPCS+Rc+1NJAlFJTvQzjMgMyRWfSy31uWiBQiNYf2OCjlD+C8-I2H1oMrSCWPLMTbIQzo0RQvlJSEh4AFp0KuTDrjKkqAEYHhw8F9FE5iYQyiBIJeV5UzXKB0BkJBuFZdkBkE4TRLzAtpQU2ERTI5QUB4mzHXynIEkcKg8jwC4AHITgQWJkF2tlKqqsBNqMnbdrOQ6YmO6qrmwtrIQ6x42J6oV9SgaNUCzMatDqqbPLtVpZqdUtXSWmITJWkpu2G5SnOQaLZXowMbG6C46roPQgbAJ81B4vB0AATTAAAqMBCCxuqADY2UJ4n0AAWW4Imnpe1o3q69jes+Uk-jWqBFUJYlkBOPAbmBuSwZciGXSgNR2BieB8XgABrOH4PRhV1aJE4JalrmaSI+XZJwL8TGiGJuN4qL5eFyAEi+H5SrAABqMBYFsGRiRZK5KbAInePd0gwFUDREFOlkEgaz3vd9-2fkDqmQ5JsOI-UFgHhN7Uzdsi3mgia3YkCX6Hby3XAgSAahryd2veJUl3bKsA68QYbStTqMjVQNkO670SE+b4045wtvB4b0Se9+3PCKkwv5q0K2ojLg0+9ND8hSdjaZC266DvgI6TrAVBHAuErJt7mNuG787Lu2vaj5PtkMHQq-YJvk075n5641YqbT6jpvqQBVMaIW1ccii2VEbaW4lZZ0iXs6MiytVbqy1qjJ2MCLhnDgXnJMBc-QK0RKvG2dsSaV2IbvZkbth5ex9gNXBAcg7p3QOPMA4djCmDwJhWO8cGFJ1oXkHubCOFcOQCYfA891REJ3EXMh68fp9yoTuXeU8OFN3AagVu7dBqdwbg1WefcB76K7gIsAuCW7x0nmYwxDxjHGhke1ORMISErxLmvW2G9b6qLcbvR+11brH3uqfd+l80TjzZL9Lw48npskCXtYJr8z4X3ql-GJd8jH-2ygmD6SCQFcWgX4PS9InZq3aIQC4NROGJyYSSY0518Z1T0DoNU2olBiXCNiTAkkCnvCKZjRwoo8w-G8UrQ4IgZDDQwDEMqVQ1CSGkFKR0G4BqwARnkP6fSBRQBSo+SWzo7w8RCDkVgfxIjkJyIbSWkAGhET2cBA5RxFjHJiKcvoFzS7jLAUbO5PNnidMUOEO6yBESgw4qAl+90-j+DyNwLQ3ZYgbDzKSdAjgPGXKUacQW28oB5AGtQBK2dEBcw-JCwZYDQWwqMginASK4go3pGijFltPFXKgNovxhgCVOGONwswgLOrAsMOgQ24LgFVhCZ8alU08iMCOUGbgBAIH5IrI6alORkkwrlQq15OR7qTjJUCsIhh8gawlf0mU+zoFnFQMGW2eZuzwu2cXHpgQabDPiHixc+JwLrk3EyoU8IlxExXhSOYiwjLhtsgZGIAB5Ww8BmBzjZQJLQOg6B0zxfDPs4zZjpugUcHiKtBhvUIRG4W4NMxss1VAaFekoC9hOYEc5OaZSpuLnWql0q3kfNKNyjVJTMXfNxMO8tlJXpqpTEKG1+LkDwDwCtSNobKjEUjQ+S28tYAbm4BrNd-wAY5AAMQAA5z2DqrLu-dAARdAY4PGFr5OYNk5gHgTvbWjbEa7t3XuQIm9AyaWwjq8YEDMRU6q4REpemUNau2jqVaGVCn8oM6Bg4YTtiifngeQ5E+OOA0PtsMCcKoIGOW7hDEVDM7sCPoc5aR2tCGoA4bANR-D0GP33AyvGJ4dwgA). You can shift-drag to create a zoom region, and when you release the mouse button the view will snap to the new scale extents. You can also double click to restore the chart to the initial configuration. Note that this works alongside standard drag-pan and scroll-zoom interactions.

The less good news is that I don't think this is currently expressible in Vega-Lite / Altair. However, I could imagine extensions to the interval selection logic that would change that. So, I'm going to rename this issue and transfer it over to the Vega-Lite repo for folks to look at there!
